### PR TITLE
*: Fix possibly invalid vector default values

### PR DESCRIPTION
**Pull request by breezewish**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/54245

Problem Summary:

In non-strict mode, 0-sized vector may be wrongly inserted into a Non-Zero-Sized vector column:

```sql
set @session.sql_mode='';
create table x(e vector(3) not null comment 'hnsw(distance=l2)');
insert into x values();
```

### What changed and how does it work?

Forbid such values in any SQL modes.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/expression/integration_test/integration_test.go [hunk 1]
```diff
@@ -60,6 +60,215 @@ import (
...
+		"<nil> <nil>",
+		"<nil> <nil>",
+	))
+	tk.MustExec("drop table t")
+
+	// ============================
+	// NULLABLE, DEFAULT
+
+	tk.MustGetErrMsg("create table t(embedding VECTOR DEFAULT '[1,2,3]')", `VECTOR column 'embedding' can't have a default value`)
+	tk.MustExec("create table t(embedding VECTOR DEFAULT (VEC_FROM_TEXT('[1,2,3]')))")
```
Comment (JaySon-Huang) [inline pkg/expression/integration_test/integration_test.go:130]: I think the error message "can't have a default value" is incorrect.

`the default value of VECTOR column 'embedding' should be defined as (VEC_FROM_TEXT(...)`?

**Review by hawkingrei — APPROVED** [review]

File: pkg/expression/integration_test/integration_test.go (hunk 1 above)
Comment (breezewish) [reply pkg/expression/integration_test/integration_test.go:130]: Good catch.. It is blindly inherited from `BLOB/TEXT/JSON column '%-.192s' can't have a default value`. We could provide a more friendly description for the VECTOR.

**Review by XuHuaiyu — APPROVED** [review]

**Review by JaySon-Huang — APPROVED** [review]
lgtm

**Pull request merged**